contracts/test/protocore/TestProtocore.sol

```diff
@@ -14,29 +14,207 @@ pragma solidity >=0.5.0 <0.6.0;
 // See the License for the specific language governing permissions and
 // limitations under the License.
 
+import "../../consensus/CoconsensusI.sol";
 import "../../protocore/Protocore.sol";
 
 contract TestProtocore is Protocore {
 
+	/* Storage */
+
     CoconsensusI public coconsensus;
 
+
     /* Special Functions */
 
     constructor(
         CoconsensusI _coconsensus,
-        uint256 genesisKernelHeight,
-        bytes32 genesisKernelHash
+        uint256 _epochLength,
+        uint256 _genesisKernelHeight,
+        bytes32 _genesisKernelHash,
+        bytes32 _genesisParentVoteMessageHash,
+        bytes32 _genesisSourceTransitionHash,
+        bytes32 _genesisSourceBlockHash,
+        bytes32 _genesisTargetBlockHash,
+        uint256 _genesisSourceBlockNumber,
+        uint256 _genesisTargetBlockNumber,
+        uint256 _genesisProposedMetablockHeight
     )
         public
     {
-        assert(genesisKernelHash != bytes32(0));
+        assert(_genesisKernelHash != bytes32(0));
+        assert(_genesisSourceBlockNumber % _epochLength == 0);
+        assert(_genesisTargetBlockHash != bytes32(0));
+        assert(_genesisTargetBlockNumber % _epochLength == 0);
+        assert(_genesisTargetBlockNumber >= _genesisSourceBlockNumber);
+        assert(_genesisProposedMetablockHeight <= _genesisKernelHeight);
+
+        openKernelHeight = _genesisKernelHeight;
+        openKernelHash = _genesisKernelHash;
+
+        bytes32 voteMessageHash = hashVoteMessageInternal(
+            _genesisSourceTransitionHash,
+            _genesisSourceBlockHash,
+            _genesisTargetBlockHash,
+            _genesisSourceBlockNumber,
+            _genesisTargetBlockNumber
+        );
+
+        Link storage link = links[voteMessageHash];
+        assert(link.targetBlockHash == bytes32(0));
+
+        link.parentVoteMessageHash = _genesisParentVoteMessageHash;
+        link.targetBlockHash = _genesisTargetBlockHash;
+        link.targetBlockNumber = _genesisTargetBlockNumber;
+        link.sourceTransitionHash = _genesisSourceTransitionHash;
+        link.proposedMetablockHeight = _genesisProposedMetablockHeight;
+        link.targetFinalisation = CheckpointFinalisationStatus.Finalised;
 
-        openKernelHeight = genesisKernelHeight;
-        openKernelHash = genesisKernelHash;
         coconsensus = _coconsensus;
+
+        super.setup(
+            _epochLength
+        );
     }
 
-    function getCoconsensus() public view returns (CoconsensusI) {
+
+    /* External Functions */
+
+    function getCoconsensus()
+		public
```

Comment:
indentation seems off

Reply:
It is correct in my VS code. Github glitch? ...